### PR TITLE
refactor downloaderTask API style

### PR DESCRIPTION
**Pull request by novain1102**
https://github.com/cocos/3d-tasks/issues/12831

### Changelog

* 

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.

  > Manual trigger with `@cocos-robot run test cases` afterward.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/native-binding/index.ts [hunk 1]
```diff
@@ -73,7 +73,68 @@ export declare namespace native {
...
+         * @en setter for the callback function while download.
+         * @zh 任务下载过程中的回调函数的修改器.
+         * @param task @en download task @zh 下载任务
+         * @param bytesReceived @en received bytes in current call @zh 此次接收到的字节
+         * @param totalBytesReceived @en total bytes have been received @zh 已接收到的所有字节
+         * @param totalBytesExpected @en total bytes expected to receive  @zh 预计接收的所有字节
+         * @example
+         * ```ts
+         *  // setter for the callback for download progress prompt
+         *  downloader.onProgress(task, bytesReceived, totalBytesReceived, totalBytesExpected) => {
```
Comment (PatriceJiang) [inline cocos/native-binding/index.ts]: missing `=`

File: cocos/native-binding/index.ts [hunk 2]
```diff
@@ -73,7 +73,68 @@ export declare namespace native {
...
+         * @en setter for the callback function when download error
+         * @zh 任务下载发生错误时的回调函数的修改器
+         * @param task @en download task @zh 下载任务
+         * @param errorCode @en  error code  @zh 错误码
+         * @param errorCodeInternal @en internal error code @zh 内部错误码
+         * @param errorStr @en error info string @zh 错误信息
+         * @example
+         * ```ts
+         * // set a download error callback
+         *  downloader.onError(task, errorCode, errorCodeInternal, errorStr) => {
```
Comment (PatriceJiang) [inline cocos/native-binding/index.ts]: `=` missing too

File: native/cocos/bindings/manual/jsb_network_manual.cpp [hunk 3]
```diff
@@ -166,20 +264,73 @@ static bool js_network_Downloader_setOnTaskError(se::State &s) { // NOLINT(reada
...
+        cobj->setOnError(arg0);
+        return true;
+    }
+    SE_REPORT_ERROR("wrong number of arguments: %d, was expecting %d", (int)argc, 1);
+    return false;
+}
+SE_BIND_FUNC_AS_PROP_SET(js_network_Downloader_setOnError)
+
 bool register_all_network_manual(se::Object * /*obj*/) {
+    __jsb_cc_network_Downloader_proto->defineProperty("onSuccess", nullptr, _SE(js_network_Downloader_setOnSuccess_asSetter));
```
Comment (PatriceJiang) [inline native/cocos/bindings/manual/jsb_network_manual.cpp:325]: Why not provide the getter, developer may access this in some cases.

Comment (PatriceJiang) [reply native/cocos/bindings/manual/jsb_network_manual.cpp:325]: It's rare in practice.

**Review by PatriceJiang — APPROVED** [review]

File: cocos/native-binding/index.ts (hunk 1 above)
Comment (novain1102) [reply cocos/native-binding/index.ts]: fixed.

File: cocos/native-binding/index.ts (hunk 2 above)
Comment (novain1102) [reply cocos/native-binding/index.ts]: fixed.

**Comment by minggo** [comment]
Will it break compatibility?

**Comment by novain1102** [comment]
> Will it break compatibility?

Compatibility work has been done. The previous interface can continue to be used

**Review by minggo — APPROVED** [review]

**Pull request merged**